src/.github/workflows/rector.yml

```
@@ -0,0 +1,68 @@
+name: Rector
+
+on:
+  pull_request:
+    branches:
+      - develop
+    paths:
+      - '**.php'
+      - 'composer.**'
+      - 'rector.php'
+      - '.github/workflows/rector.yml'
+  push:
+    branches:
+      - develop
+    paths:
+      - '**.php'
+      - 'composer.**'
+      - 'rector.php'
+      - '.github/workflows/rector.yml'
+
+jobs:
+  build:
+    name: PHP ${{ matrix.php-versions }} Rector Analysis
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, '[ci skip]')"
+    strategy:
+      fail-fast: false
+      matrix:
+        php-versions: ['7.3', '7.4', '8.0']
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-versions }}
+          tools: phpstan
+          extensions: intl, json, mbstring, xml
+          coverage: none
+        env:
+          COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Get composer cache directory
+        id: composer-cache
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+      - name: Cache composer dependencies
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}-${{ hashFiles('**/composer.lock') }}
+          restore-keys: ${{ runner.os }}-composer-
+
+      - name: Install dependencies
+        run: |
+          composer -q config -g github-oauth.github.com "${{ secrets.GITHUB_TOKEN }}"
+          if [ -f composer.lock ]; then
+            composer install --no-progress --no-interaction --prefer-dist --optimize-autoloader
+          else
+            composer update --no-progress --no-interaction --prefer-dist --optimize-autoloader
+          fi
+
+      - name: Analyze for refactoring
+        run: |
+          composer global require --dev rector/rector:^0.12.4
```

better to have pinned version so we can verify every version to verify bug fix, like in https://github.com/codeigniter4/CodeIgniter4/pull/5328#issuecomment-969483172

```suggestion
          composer global require --dev rector/rector:0.12.4
```

---

I'm not sure how to handle this one. The goal is for these to be templates that developers can merge into their own repos and then adjust as necessary. I don't want us to have to keep updating this repo every time there is a new minor patch to Rector. Maybe we reference the minor version `^0.12` and add a note to the docs that developers should consider locking it to the specific patch?

---

I rolled it back to the unspecified version and added a note in the docs; see what you think.